frontend/lib/auditlogquery/index.ts

```diff
@@ -0,0 +1,200 @@
+        })
+        break
+      case 'ActorOwnerId':
+        result.push({
+          inActorOwnerId: value.split('|') as AuditLogQueryWhere['inActorOwnerId'],
+        })
+        break
+      case 'Id':
+        result.push({
+          inId: value.split('|') as AuditLogQueryWhere['inId'],
+        })
+        break
+      case 'TargetType':
+        result.push({
+          inTargetType: value.split('|') as AuditLogQueryWhere['inTargetType'],
+        })
+        break
+      case 'TargetId':
+        result.push({
+          inTargetId: value.split('|') as AuditLogQueryWhere['inTargetId'],
+        })
+        break
+      case 'TargetOwnerId':
+        result.push({
+          inTargetOwnerId: value.split('|') as AuditLogQueryWhere['inTargetOwnerId'],
+        })
+        break
+      case 'MinCreatedAt':
+        result.push({
+          minCreatedAt: value,
+        })
+        break
+      case 'MaxCreatedAt':
+        result.push({
+          maxCreatedAt: value,
+        })
+        break
+      default:
+        console.warn(new Error(`Unknown where: ${JSON.stringify(key)}`))
+    }
+  }
+  return result
+}
+
+const encodeAuditLogQueryLimit = (limit: number): string => {
+  if (limit === limitDefault) {
+    return ''
+  }
+  return `${limit}`
+}
+
+const decodeAuditLogQueryLimit = (limit: string): number => {
+  if (limit === '') {
+    return limitDefault
+  }
+  return parseInt(limit)
+}
+
+const encodeAuditLogQueryCursor = (cursor: string): string => {
+  return encodeURIComponent(cursor)
+}
+
+const decodeAuditLogQueryCursor = (cursor: string): string => {
+  return decodeURIComponent(cursor)
+}
+
+const sortsQP = 's'
+const wheresQP = 'w'
+const limitQP = 'l'
+const limitDefault = 100
+const cursorQP = 'c'
+const pageURLBase = '/auditlog'
+
+export const urlReactiveAuditLogQuery = (fromQueryReactiveWithDefault: (key: string, defaultValue: string) => WritableComputedRef<string>): WritableComputedRef<AuditLogQueryReq> => {
+  const qSorts = fromQueryReactiveWithDefault(sortsQP, '')
+  const qWheres = fromQueryReactiveWithDefault(wheresQP, '')
+  const qLimit = fromQueryReactiveWithDefault(limitQP, '')
+  const qCursor = fromQueryReactiveWithDefault(cursorQP, '')
+
+  return computed({
+    get: (): AuditLogQueryReq => {
+      return {
+        sorts: decodeAuditLogQuerySorts(qSorts.value),
+        wheres: decodeAudtLogQueryWheres(qWheres.value),
+        limit: decodeAuditLogQueryLimit(qLimit.value),
+        cursor: decodeAuditLogQueryCursor(qCursor.value),
+      }
+    },
+    set: (value: AuditLogQueryReq) => {
+      qSorts.value = encodeAuditLogQuerySorts(value.sorts ?? [])
+      qWheres.value = encodeAuditLogQueryWheres(value.wheres)
+      qLimit.value = encodeAuditLogQueryLimit(value.limit ?? limitDefault)
+      qCursor.value = encodeAuditLogQueryCursor(value.cursor ?? '')
+    },
+  })
+}
+
+export const createURLAuditLogQuery = (localePath: ReturnType<typeof useLocalePath>, req: AuditLogQueryReq): string => {
```

Comment:
Looks like `ReturnType<typeof useLocalePath>` can just be `LocalePathFunction`? At least if I followed that chain right

Reply:
Great catch